workflow/controller/dag_test.go

```diff
@@ -3448,3 +3448,178 @@ func TestDagHttpChildrenAssigned(t *testing.T) {
+      name: ""
+      resources: {}
+    inputs: {}
+    metadata: {}
+    name: intentional-fail
+    outputs: {}
+status:
+  artifactRepositoryRef:
+    artifactRepository:
+      archiveLogs: true
+      s3:
+        accessKeySecret:
+          key: accesskey
+          name: my-minio-cred
+        bucket: my-bucket
+        endpoint: minio:9000
+        insecure: true
+        secretKeySecret:
+          key: secretkey
+          name: my-minio-cred
+    configMap: artifact-repositories
+    key: default-v1
+    namespace: argo
+  conditions:
+  - status: "False"
+    type: PodRunning
+  - status: "True"
+    type: Completed
+  finishedAt: "2022-07-18T22:45:40Z"
+  nodes:
+    dag-contiue-on-fail-987nb:
+      children:
+      - dag-contiue-on-fail-987nb-3791482269
+      - dag-contiue-on-fail-987nb-3741149412
+      displayName: dag-contiue-on-fail-987nb
+      finishedAt: "2022-07-18T22:45:40Z"
+      id: dag-contiue-on-fail-987nb
+      name: dag-contiue-on-fail-987nb
+      outboundNodes:
+      - dag-contiue-on-fail-987nb-3791482269
+      - dag-contiue-on-fail-987nb-3741149412
+      phase: Failed
+      progress: 1/2
+      resourcesDuration:
+        cpu: 8
+        memory: 4
+      startedAt: "2022-07-18T22:45:27Z"
+      templateName: workflow
+      templateScope: local/dag-contiue-on-fail-987nb
+      type: DAG
+    dag-contiue-on-fail-987nb-3741149412:
+      boundaryID: dag-contiue-on-fail-987nb
+      displayName: A
+      finishedAt: "2022-07-18T22:45:39Z"
+      hostNodeName: k3d-k3s-default-server-0
+      id: dag-contiue-on-fail-987nb-3741149412
+      name: dag-contiue-on-fail-987nb.A
+      outputs:
+        artifacts:
+        - name: main-logs
+          s3:
+            key: dag-contiue-on-fail-987nb/dag-contiue-on-fail-987nb-whalesay-3741149412/main.log
+        exitCode: "0"
+      phase: Succeeded
+      progress: 1/1
+      resourcesDuration:
+        cpu: 4
+        memory: 2
+      startedAt: "2022-07-18T22:45:33Z"
+      templateName: whalesay
+      templateScope: local/dag-contiue-on-fail-987nb
+      type: Pod
+    dag-contiue-on-fail-987nb-3791482269:
+      boundaryID: dag-contiue-on-fail-987nb
+      displayName: B
+      finishedAt: "2022-07-18T22:45:31Z"
+      hostNodeName: k3d-k3s-default-server-0
+      id: dag-contiue-on-fail-987nb-3791482269
+      message: Error (exit code 1)
+      name: dag-contiue-on-fail-987nb.B
+      outputs:
+        artifacts:
+        - name: main-logs
+          s3:
+            key: dag-contiue-on-fail-987nb/dag-contiue-on-fail-987nb-intentional-fail-3791482269/main.log
+        exitCode: "1"
+      phase: Failed
+      progress: 0/1
+      resourcesDuration:
+        cpu: 4
+        memory: 2
+      startedAt: "2022-07-18T22:45:27Z"
+      templateName: intentional-fail
+      templateScope: local/dag-contiue-on-fail-987nb
+      type: Pod
+  phase: Failed
+  progress: 1/2
+  resourcesDuration:
+    cpu: 8
+    memory: 4
+  startedAt: "2022-07-18T22:45:27Z"
```

Comment:
I think we can remove all these static `status` field
and run `woc.operate(ctx)` twice to achieve the same purpose

Reply:
Hi @dpadhiar , it won't mark the workflow as complete

---

here's how it works in my head:
- 1st operate, executes pod B due priority
- 2nd operate, executes pod A due priority

Then we can compare the `startedAt`

At this stage, the workflow is at phase.status running, and I think this is all we need.

---

Giving it the status with phase Failed won't cause operate() to perform any further action, we will end up evaluating the test data instead of the controller functionality

---

Let me know what you think :)

Reply:
@tczhao This implementation worked exactly as you said! Thank you for the help, ready to add.